korman/properties/modifiers/anim.py

```diff
@@ -60,37 +60,44 @@ class PlasmaAnimationModifier(ActionModifier, PlasmaModifierProperties):
                                 description="Marker indicating where the default loop begins")
     loop_end = StringProperty(name="Loop End",
                               description="Marker indicating where the default loop ends")
+    obj_sdl_anim = StringProperty(name="SDL Animation",
+                                  description="Name of the SDL variable to use for this animation",
+                                  options=set())
 
     def export(self, exporter, bo, so):
         action = self.blender_action
 
-        atcanim = exporter.mgr.find_create_object(plATCAnim, so=so)
-        atcanim.autoStart = self.auto_start
-        atcanim.loop = self.loop
-
-        # Simple start and loop info
-        if action is not None:
-            markers = action.pose_markers
-            initial_marker = markers.get(self.initial_marker)
-            if initial_marker is not None:
-                atcanim.initial = _convert_frame_time(initial_marker.frame)
-            else:
-                atcanim.initial = -1.0
-            if self.loop:
-                loop_start = markers.get(self.loop_start)
-                if loop_start is not None:
-                    atcanim.loopStart = _convert_frame_time(loop_start.frame)
+        anim_data = bo.plasma_modifiers.animation
+        if anim_data.obj_sdl_anim:
+            atcanim = exporter.mgr.find_create_object(plAgeGlobalAnim, so=so)
+        else:
+            atcanim = exporter.mgr.find_create_object(plATCAnim, so=so)
```

Comment:
This pattern of checking `obj_sdl_anim` to figure out what kind of object to create is happening fairly often. We could hide this detail by  adding a property to the animation modifier, like so:
```python
@property
def anim_type(self):
    return plAgeGlobalAnim if self.obj_sdl_anim else plATCAnim
```
Then you could simplify some of the export code by doing things like:
```python
anim = mgr.find_create_object(anim_mod.anim_type, so=so)
if isinstance(anim, plAgeGlobalAnim):
    # ... code for age SDL animation here ...
else:
    # ... regular animation but not SDL animation here ...
# both here...
```

Reply:
Done!